elyra/pipeline/validation.py

```diff
@@ -217,25 +217,27 @@ def _validate_pipeline_structure(self, pipeline_definition: PipelineDefinition,
 
         # validate pipeline version compatibility
-        pipeline_version = int(pipeline_definition.primary_pipeline.version)
-        if pipeline_version not in range(PIPELINE_CURRENT_VERSION + 1):
-            response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
-                                 message="Primary pipeline version field has an invalid value.",
-                                 data={"supported_version": PIPELINE_CURRENT_VERSION,
-                                       "detected_version": pipeline_version})
-
-        elif pipeline_version < PIPELINE_CURRENT_VERSION:
-            # Pipeline needs to be migrated
-            response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
-                                 message=f'Pipeline version {pipeline_version} is out of date and needs to be migrated '
-                                         f'using the Elyra pipeline editor.')
-        elif pipeline_version > PIPELINE_CURRENT_VERSION:
-            # New version of Elyra is needed
-            response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
-                                 message='Pipeline was last edited in a newer version of Elyra. '
-                                         'Update Elyra to use this pipeline.',
-                                 data={"supported_version": PIPELINE_CURRENT_VERSION,
-                                       "detected_version": pipeline_version})
+        try:
+            pipeline_version = pipeline_definition.primary_pipeline.version
+            if pipeline_version < PIPELINE_CURRENT_VERSION:
+                # Pipeline needs to be migrated
+                response.add_message(severity=ValidationSeverity.Error,
+                                     message_type="invalidPipeline",
+                                     message=f'Pipeline version {pipeline_version} is '
+                                             'out of date and needs to be migrated '
+                                             f'using the Elyra pipeline editor.')
+            elif pipeline_version > PIPELINE_CURRENT_VERSION:
+                # New version of Elyra is needed
+                response.add_message(severity=ValidationSeverity.Error,
+                                     message_type="invalidPipeline",
+                                     message='Pipeline was last edited in a newer version of Elyra. '
+                                             'Update Elyra to use this pipeline.',
+                                     data={"supported_version": PIPELINE_CURRENT_VERSION,
+                                           "detected_version": pipeline_version})
+        except ValueError:
+            response.add_message(severity=ValidationSeverity.Error,
+                                 message_type="invalidPipeline",
+                                 message='Pipeline version is not a numeric value.')
```

Comment:
During a non-CLI submission, I don't think this will ever be reached since we convert to `int` [here](https://github.com/elyra-ai/elyra/blob/e628ffc4948ceaf32e2a0873c4164c4956c13656/elyra/pipeline/pipeline_definition.py#L90). That's obviously not an error, but just wondering if we would want to remove the conversion for consistency across methods of submission (assuming there's not something I'm overlooking in this aspect)

Reply:
I believe it is, because that type casting will raise the `ValueError` for non-numeric values.

Reply:
Python won't raise a `ValueError` for something like `int("5")` (it will return 5), but you're right that it will for a string that can't be cast into a simple integer, e.g. `int("k")` or `int("2.2")`

Reply:
Right, that that was the intention here: the `except` covers those cases where the pipeline file contains a non-integer version string. Not that I have ever seen a file with on , but that doesn't mean in can't happen in the future (:

Reply:
Ah got it :)